enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -430,12 +402,44 @@ type PlatformSpec struct {
 }
 ```
 
-For the sake of consistency, status should be introduced as well. For now, this will be empty.
+For the sake of consistency, status should be introduced as well, and will
+define the settings set at installation time:
 
 ```go
+type CloudControllerManagerMode string
+
+const (
+    // Cloud Controller Manager is enabled and expected to be supplied.
+    // Signaling that kubelets and other CCM consumers should use --cloud-provider=external.
+    CloudControllerManagerExternal CloudControllerManagerMode = "External"
+    // Cloud Controller Manager is enabled and expected to be supplied.
+    // Signaling that kubelets and other CCM consumers should not set --cloud-provider flag.
+    CloudControllerManagerNone CloudControllerManagerMode = "None"
+)
+
+// CloudControllerManagerStatus holds the state of Cloud Controller Manager (a.k.a. CCM or CPI) related settings
+type CloudControllerManagerStatus struct {
+    // state determines whether or not an external Cloud Controller Manager is expected to
+    // be installed within the cluster.
+    // https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager
+    //
+    // When set to "External", new nodes will be tainted as uninitialized when created,
+    // preventing them from running workloads until they are initialized by the cloud controller manager.
+    // When omitted or set to "None", new nodes will be not tainted
+    // and no extra initialization from the cloud controller manager is expected.
+    // +kubebuilder:default:=""
+    // +default=""
+    // +kubebuilder:validation:Enum="";External;None
+    // +optional
+    State CloudControllerManagerState `json:"state"`
```

Comment:
CEL validation to make this immutable once set will be required

Reply:
done